lib/cn/cn_tree.c

```diff
@@ -1589,9 +1623,17 @@ cn_comp_update_split(
             cn_tree_samp(tree, &w->cw_samp_post);
         }
 
+        /* Advance the split generation count to ensure that all spill operations
+         * that overlap this split will abort and retry.
+         */
+        tree->ct_splitgen++;
+
         atomic_sub_rel(&right->tn_busycnt, (1u << 16) + w->cw_kvset_cnt);
     } while (0);
 
+    /* TODO: If we encountered an error then we must restore the node to its pre-split
```

Comment:
Thanks for catching this! That's my cleanup bug, will get it fixed.

All error paths in this function result in a KVDB health error which halts the KVDB, so this restoration path didn't get my attention.

Reply:
Please do not do that.  Please allocate resources before the CNDB ACKs.